src/source/PeerConnection/SessionDescription.c

```diff
@@ -316,6 +327,74 @@ PCHAR fmtpForPayloadType(UINT64 payloadType, PSessionDescription pSessionDescrip
     return NULL;
 }
 
+/*
+ * Extracts a (hex) value after the provided prefix string. Returns true if
+ * successful.
+ */
+BOOL readHexValue(PCHAR input, PCHAR prefix, PINT32 value)
+{
+    PCHAR substr = STRSTR(input, prefix);
+    if (substr != NULL) {
+        // TODO: This should be changed to SSCANF once the PIC library macro
+        // change is propagated and available here.
+        if (sscanf(substr + STRLEN(prefix), "%x", value) == 1) {
+            return TRUE;
+        }
+    }
+    return FALSE;
+}
+
+/*
+ * Scores the provided fmtp string based on this library's ability to
+ * process various types of H264 streams. A score of 0 indicates an
+ * incompatible fmtp line. Beyond this, a higher score indicates more
+ * compatibility with the desired characteristics, packetization-mode=1,
+ * level-asymmetry-allowed=1, and inbound match with our preferred
+ * profile-level-id. Scoring is on a scale from [0,1], with 1 deemed as
+ * a "perfect fit".
+ *
+ * At some future time, it may be worth expressing this as a true distance
+ * function as defined here, although dealing with infinite floating point
+ * values can get tricky:
+ * https://www.w3.org/TR/mediacapture-streams/#dfn-fitness-distance
```

Comment:
I really don't understand why this is any better than simply having a cumulative sum like we are doing

Reply:
The linked document uses a cumulative sum to add "features".